### PR TITLE
[core] [6/N] Allow users to specify uv version to install

### PR DESCRIPTION
**Pull request by dentiny**
Followup PR on https://github.com/ray-project/ray/pull/48611, this PR allows user to user a specific version of `uv`.

I think I did a bad thing: I push all my commits into `dentiny/ray` instead of `ray-project/ray`, which makes it hard to compare base branch across two repositories :(

The only commit to review is: https://github.com/ray-project/ray/pull/48634/commits/6d2092ec7f33cc2a0ac7ae4cbe21647b6cf79efe

### CODE REVIEW

**Comment by dentiny** [comment]
> Rebase.

Rebased!

File: python/ray/tests/test_runtime_env_uv.py [hunk 1]
```diff
@@ -69,6 +69,19 @@ def f():
...
 
+# Specify uv version.
+def test_uv_with_version(shutdown_only):
+    @ray.remote(
+        runtime_env={"uv": {"packages": ["requests==2.3.0"], "uv_version": "==0.4.0"}}
+    )
+    def f():
+        import requests
+
+        return requests.__version__
```
Comment (jjyao) [inline python/ray/tests/test_runtime_env_uv.py:80]: Can we also do something like

```
import uv
uv.__version__
```
to make sure we install the right version of uv

File: python/ray/_private/runtime_env/uv.py [hunk 2]
```diff
@@ -79,14 +79,22 @@ async def _install_uv(
         virtualenv_path = virtualenv_utils.get_virtualenv_path(path)
         python = virtualenv_utils.get_virtualenv_python(path)
```
Comment (jjyao) [inline python/ray/_private/runtime_env/uv.py:81]: After 5/N the high level logic should be

```
if uv_version is not None:
  pip install --ignore-installed uv=xxxx
else:
  if uv is installed:
    do nothing
  else:
    installed the latest
```

?

File: python/ray/tests/test_runtime_env_uv.py (hunk 1 above)
Comment (dentiny) [reply python/ray/tests/test_runtime_env_uv.py:80]: Sounds good! Updated.

File: python/ray/_private/runtime_env/uv.py (hunk 2 above)
Comment (dentiny) [reply python/ray/_private/runtime_env/uv.py:83]: Sounds good, btw did some quick research, seems `--force-reinstall` is preferred.
Eg: https://stackoverflow.com/questions/51913361/difference-between-pip-install-options-ignore-installed-and-force-reinstall

Comment (dentiny) [reply python/ray/_private/runtime_env/uv.py:85]: Updated.

File: python/ray/_private/runtime_env/validation.py [hunk 3]
```diff
@@ -149,16 +148,21 @@ def parse_and_validate_uv(uv: Union[str, List[str], Dict]) -> Optional[Dict]:
...
+                "packages and uv_version, but got: "
                 f"{list(uv.keys())}"
             )
         if "packages" not in uv:
             raise ValueError(
                 f"runtime_env['uv'] must include field 'packages', but got {uv}"
             )
+        if "uv_version" in uv and not isinstance(uv["uv_version"], str):
+            raise TypeError(
+                "runtime_env['uv']['uv_version'] must be of type bool, "
```
Comment (jjyao) [inline python/ray/_private/runtime_env/validation.py:163]:
```suggestion
                "runtime_env['uv']['uv_version'] must be of type str, "
```

File: python/ray/tests/test_runtime_env_uv.py [hunk 4]
```diff
@@ -69,6 +69,32 @@ def f():
...
+def test_uv_with_version_and_check(shutdown_only):
+    @ray.remote(
+        runtime_env={"uv": {"packages": ["requests==2.3.0"], "uv_version": "==0.4.0"}}
+    )
+    def f():
+        import pkg_resources
+
+        return pkg_resources.get_distribution("uv").version
+
+    assert ray.get(f.remote()) == "0.4.0"
```
Comment (jjyao) [inline python/ray/tests/test_runtime_env_uv.py:95]: We just need one test

```
def f():
   import requests
   import pkg_resources
   
   assert requests.__version__ == "2.3.0"
   assert pkg_resources.get_distribution("uv").version == "0.4.0"
```

Comment (dentiny) [reply python/ray/tests/test_runtime_env_uv.py:95]: Updated.

**Review by jjyao — APPROVED** [review]

**Pull request merged**